app/facades/search_results_facade.rb

```diff
@@ -0,0 +1,63 @@
+class SearchResultsFacade
+  def initialize
+    @service ||= MsetService.new
+  end
+
+  def med_search_results(name)
+    response = @service.med_search(name)
+    results = json_parse(response)[:results]
+    return nil if results.nil?
+    med_and_ndc_hash(results)
+  end
+
+  def extract_symptoms(product_ndc)
+    tables = adverse_reactions_table(product_ndc)
+    return nil if tables[0].nil?
+    symptoms = []
+    tables.each do |table|
+      table.each do |t|
+        nokogiri_parser(t, symptoms)
+      end
+    end
+    symptoms.uniq!
+    symptoms.delete("") if symptoms.include?("")
+    symptoms
+  end
+
+  private
+
+  def adverse_reactions_table(product_ndc)
+    response = @service.sym_search(product_ndc)
+    results = json_parse(response)[:results]
+    results.map do |result|
+      result[:adverse_reactions_table]
+    end
+  end
+
+  def nokogiri_parser(table, acc)
+    page = Nokogiri::XML(table)
+    page.css('tbody').select do |node|
+      node.traverse do |el|
+        acc << el.text.strip unless el.text.include?('%') || el.text.include?('System') || el.text.strip == 'General' || el.text.strip == 'Metabolic/Nutritional' || el.name == 'footnote' || el.text == ' ' || el.text.split(' ').size > 3 || el.text.include?('only') || el.text=~ /\d/
+      end
+    end
+  end
+
+  def med_and_ndc_hash(results)
+    # should this return a temporary_medication PORO
```

Comment:
i think ideally we should create a PORO here so that we're passing objects to the view? however, this could get messy since we have a Medication resource. i'm open to differences of opinion here as this is currently working - certainly not a top priority

Reply:
Yeah, if anything, if we decide to create a PORO for it, i would just vote not to call it 'medications'.. but seeing as this hypothetical object wouldn't get passed around much... not sure how much it's worth doing all that.